pkg/hsskey/filereader.go

```diff
@@ -156,3 +158,74 @@ func getHssFromFile(verbose bool, verboseDangerous bool, filePaths []string) ([]
 
 	return allHss, nil
 }
+
+// WriteHssToTempFile writes a list of HSS to a tmpfs file where the filepath is returned.
+// See WriteHssToFile for HSS details.
+func WriteHssToTempFile(warnings io.Writer, verboseDangerous bool, hss [][]byte) (string, error) {
+	file, err := os.CreateTemp("", "hss_*")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp file %v", err)
+	}
+
+	err = WriteHssToFile(warnings, verboseDangerous, file, hss)
+	if err != nil {
+		file.Close()
+		os.Remove(file.Name())
+		return "", err
+	}
+
+	file.Close()
+	return file.Name(), nil
+}
+
+// WriteHssToFile writes a list of HSS to an open file.
+// Each HSS key is expected to be 64 bytes long and has a checksum for validation.
+// HSS that fail to validate are not written to the file.
+// Function returns error if no HSS are written or writing can no longer continue.
+func WriteHssToFile(warnings io.Writer, verboseDangerous bool, file *os.File, hss [][]byte) error {
+	if len(hss) == 0 {
+		return fmt.Errorf("InvalidArgument: No HSS to write")
+	}
+	if file == nil {
+		return fmt.Errorf("InvalidArgument: file is nil")
+	}
+	// Count of successful writes
+	written := 0
+
+	for i := 0; i < len(hss); i++ {
+		hssKey := hss[i]
+
+		// Verify the checksums.
+		if !validateChecksum(hssKey) {
+			if warnings != nil {
+				msg := fmt.Sprintf("Checksum validation failed for HSS key index %d", i)
+				if verboseDangerous {
+					msg = msg + fmt.Sprintf("\nseed=%x, seed(octal escape sequence)=%s", hssKey,
+						toOctalEscapeSequence(hssKey))
+				}
+				io.WriteString(warnings, msg)
+			}
+			continue
+		}
+
+		if verboseDangerous && warnings != nil {
+			msg := fmt.Sprintf("Writing HSS key to file=%s\nseed=%x, seed(octal escape sequence)=%s",
+				file.Name(), hssKey, toOctalEscapeSequence(hssKey))
+			io.WriteString(warnings, msg)
+		}
+
+		n, err := file.Write(hssKey)
+		if n != len(hssKey) {
+			return fmt.Errorf("Failed to write entire HSS key to file %s", file.Name())
```

Comment:
Don't capitalize error strings, here and everywhere else in the file: https://google.github.io/styleguide/go/decisions#error-strings

Reply:
Thanks, fixed. Capitalizing all io.WriteString messages and lowercases all fmt.Error.